### PR TITLE
feature/dashboard/add_taxi_table

### PR DESCRIPTION
**Pull request by Asiasama710**

### CODE REVIEW

**Review by devsadeq — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+private val colorMapper: Map<Int, Color> = mapOf(
+    TaxiColor.RED.colorNumber to Color(0xFFF47373),
+    TaxiColor.YELLOW.colorNumber to Color(0xFFF8EC7E),
+    TaxiColor.GREEN.colorNumber to Color(0xFF80E5AB),
+    TaxiColor.BLUE.colorNumber to Color(0xFF77DEEE),
+    TaxiColor.WHITE.colorNumber to Color.White,
+    TaxiColor.GRAY.colorNumber to Color(0xFFAFAFAF),
+    TaxiColor.BLACK.colorNumber to Color(0xFF3F3F3F),
+)
+private val statusMapper: Map<Int, TaxiStatus> = mapOf(
```
Comment (mosayed01) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:29]: Please rename it to status map. It was confusing to me.

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    plateNumber = plateNumber,
+    color = colorMapper[color] ?: Color.White,
+    type = type,
+    seats = seats,
+    status = statusMapper[status]?: TaxiStatus.ONLINE,
+    username = username,
+    trips = trips,
+)
+
+private val colorMapper: Map<Int, Color> = mapOf(
```
Comment (mosayed01) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:20]: Please rename it to colorMap

**Review by mosayed01 — APPROVED** [review]
Please don't forget to rename mentioned variables

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    TaxiColor.RED.colorNumber to Color(0xFFF47373),
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:21]: It Should be better to get these colors from Local composition from our Theme , if not existed you shoud add them

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    TaxiColor.YELLOW.colorNumber to Color(0xFFF8EC7E),
+    TaxiColor.GREEN.colorNumber to Color(0xFF80E5AB),
+    TaxiColor.BLUE.colorNumber to Color(0xFF77DEEE),
+    TaxiColor.WHITE.colorNumber to Color.White,
+    TaxiColor.GRAY.colorNumber to Color(0xFFAFAFAF),
+    TaxiColor.BLACK.colorNumber to Color(0xFF3F3F3F),
+)
+private val statusMapper: Map<Int, TaxiStatus> = mapOf(
+    0 to TaxiStatus.OFFLINE,
+    1 to TaxiStatus.ONLINE,
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:31]: Hard coded numbers

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/util/TaxiColor.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package org.thechance.common.domain.util
+
+
+enum class TaxiColor(val colorNumber: Int) {
+    RED(0),
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/util/TaxiColor.kt:5]: I think color numbers is also hard coded

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/table/TotalItemsIndicator.kt [hunk 6]
```diff
@@ -21,25 +21,24 @@ import com.beepbeep.designSystem.ui.theme.Theme
...
     totalItems: Int,
+    numberItemInPage: String,
     onItemPerPageChange: (String) -> Unit,
-    itemType: String
+    itemType: String,
 ) {
     Row(
         modifier = modifier,
         verticalAlignment = Alignment.CenterVertically,
         horizontalArrangement = Arrangement.spacedBy(12.dp)
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/table/TotalItemsIndicator.kt:32]: I thing we have local composition for dimens , you should use it

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/table/TotalItemsIndicator.kt [hunk 7]
```diff
@@ -21,25 +21,24 @@ import com.beepbeep.designSystem.ui.theme.Theme
...
     ) {
-
         BasicTextField(
             modifier = Modifier
                 .border(
                     width = 1.dp,
                     color = Theme.colors.contentBorder,
                     shape = RoundedCornerShape(8.dp)
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/table/TotalItemsIndicator.kt:39]: Also here , you should use our local composition for radius

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiScreen.kt [hunk 8]
```diff
@@ -1,42 +1,230 @@
...
+    @Composable
+    private fun RowScope.TaxiRow(
+        onClickEdit: (id: String) -> Unit,
+        position: Int,
+        taxi: TaxiDetailsUiState,
+        firstColumnWeight: Float = 1f,
+        otherColumnsWeight: Float = 3f,
+    ) {
+        Text(
+            text = position.toString(),
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiScreen.kt:158]: This text repeated many times , find a way to reduce duplicating

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiUiState.kt [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+) {
+    val statusColor: Color
+        @Composable get() = when (status) {
+            TaxiStatus.OFFLINE -> Theme.colors.primary
+            TaxiStatus.ONLINE -> Theme.colors.success
+            TaxiStatus.ON_RIDE -> Theme.colors.warning
+        }
+    val statusText: String
+        get() = when (status) {
+            TaxiStatus.OFFLINE -> "Offline"
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiUiState.kt:50]: Hard coded Strings

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpButton.kt [hunk 10]
```diff
@@ -31,6 +32,7 @@ fun BpButton(
     modifier: Modifier = Modifier,
     painter: Painter? = null,
     enabled: Boolean = true,
+    textPadding: PaddingValues = PaddingValues(Theme.dimens.space16),
     shape: Shape = RoundedCornerShape(Theme.radius.medium),
     containerColor: Color = Theme.colors.primary,
     contentColor: Color = Color.White,
```
Comment (ahmednasserzaza) [inline design_system/shared/src/commonMain/kotlin/ui/composable/BpButton.kt:38]: you should use local composition to get this color and add it if not existed

**Review by ahmednasserzaza — APPROVED** [review]
Nice Work 🌺

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt (hunk 3 above)
Comment (Asiasama710) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:21]: but this in data layer and mapping to domain layer , i can't use Local composition in here

**Pull request merged**